src/main.rs

```diff
@@ -19,7 +19,13 @@ fn serve_static_file(mut stream: TcpStream, path: &str) {
     };
     let mut buffer = Vec::new();
     file.read_to_end(&mut buffer).expect("Read failed");
-
+    let mut parts = path.split('.').collect::<Vec<&str>>();
+    let header:String = match parts.pop() {
+        Some("pdf") => format!("HTTP/1.1 200 OK\r\nContent-Type: application/pdf\r\nContent-Disposition:inline;filename={}\r\n\r\n", path),
+        Some(_) => "HTTP/1.1 200 OK\r\nContent-Type: text/html; charset=UTF-8\r\n\r\n".to_string(),
```

Comment:
You should extend the match to include different file types.  For javascript ("js") you should change the `Content-Type` to `application/javascript`, for CSS ("css"), it should be `text/css`.  There are many more, which is why I stopped at resolving the issue indicated in the issue.  This could easily become an entire module on its own.